### DISCUSSION TITLE
S7 protocol?  Profinet protocol?

### DISCUSSION THREAD

**Discussion by luewh** [gijzelaerr/python-snap7#447] [Q&A] [answered]
**Is S7 comm a profinet protocol ?**

I ask this question because someone told me do not use "Snap7 protocol" to communicate between PLC(1500) and PC, use PROFINET instead.

But I heard that Profinet is the standard protocol for S1200 and S1500 PLC and I succeed to change data between PLC and PC by using python-snap7, if snap7 isn't Profinet protocol witch protocol it is then ? S7 comm ? Yes, but where it goes Profinet ??? I'm so confused...

I have a this link talks about it but I can't understand : https://www.promotic.eu/en/pmdoc/Subsystems/Comm/PLC/Simatic/Simatic.htm

extra question : Is it possible to do a RT Profinet communication between PLC-PC ? With python-snap7 ?
![profinet](https://user-images.githubusercontent.com/85931693/236209593-ebe96055-a2fe-4622-bb18-d4fd5384e34c.PNG)
![s7](https://user-images.githubusercontent.com/85931693/236209617-46b4ae80-4c6c-4aaa-9dfa-73118fa95c3c.PNG)

**Comment by nikteliy** [comment, marked as answer]
Hi @luewh 
> Is S7 comm a Profinet protocol ?

No, s7comm and Profinet are two independent protocols.
Both protocols can be used over Ethernet(via a Profinet(LAN) **interface** on your PLC). 

> Is it possible to do a RT Profinet communication between PLC-PC ?

Yes. You'll most likely need additional software such as a Profinet driver, configuration tool etc.
But I have never tried it myself, so I cannot be certain.

> With python-snap7 ?

No. Python-snap7 supports only s7comm protocol.
